lib/actions/ui.js

```diff
@@ -202,37 +204,69 @@ export function moveTo(i) {
   };
 }
 
+function getDefaultJsEditor() {
+  const regKey = '\\JSFile\\Shell\\Edit\\Command';
+  new Registry({
+    hive: 'HKCR',
+    key: regKey
+  }).get(Registry.DEFAULT_VALUE, (err, entry) => {
+    if (err) {
+      // In case of error, return default editor
+      return 'notepad.exe';
+    }
+    // Default is "C:\Windows\System32\Notepad.exe %1"
+    entry = entry.substring(0, entry.indexOf(' %1'));
+    entry = entry.substring(entry.lastIndexOf('\\'), entry.length);
+    return entry;
+  });
+}
+
 export function showPreferences() {
   const isWin = process.platform === 'win32';
   // eslint-disable-next-line no-template-curly-in-string
-  const command = isWin ? ' start notepad "%userprofile%\\.hyper.js"' : ' bash -c \'exec env ${EDITOR:=nano} ~/.hyper.js\'';
-  const message = [];
-  message.push(isWin ?
-    ' echo Attempting to open ^%userprofile^%\\.hyper.js with notepad' :
-    ' echo Attempting to open ~/.hyper.js with your \\$EDITOR');
-  message.push(' echo If it fails, open it manually with your favorite editor!');
+  let command = ' bash -c \'exec env ${EDITOR:=nano} ~/.hyper.js\'';
+  let handled = false;
 
-  return dispatch => {
-    dispatch({
-      type: UI_SHOW_PREFERENCES,
-      effect() {
-        dispatch(requestSession());
-        // Replace this hack with an async action
-        rpc.once('session add', ({uid}) => {
-          rpc.once('session data', () => {
-            dispatch(sendSessionData(
-              uid,
-              [
-                ...message,
-                command,
-                ''
-              ].join(EOL)
-            ));
+  if (isWin) {
+    const jsEditor = getDefaultJsEditor();
+    if (!jsEditor || jsEditor.toLowerCase() === 'notepad.exe') {
+      handled = true;
+      shell.showItemInFolder('%userprofile%\\.hyper.js');
+    } else {
+      // eslint-disable-next-line no-template-curly-in-string
+      command = ' start ${jsEditor} "%userprofile%\\.hyper.js"';
+    }
+  }
+
+  if (!handled) {
```

Comment:
You should handle code in this if block in a callback since line 241 may happen before `getDefaultJsEditor()` has returned anything as `winreg` is async.